### PR TITLE
Go to file resource if are relationship type

### PR DESCRIPTION
**Pull request by andresilvagomez**

### CODE REVIEW

**Comment by ahmedkandel** [comment]
Thanks, good addition but please keep in mind that not every relationship has a corresponding resource. Maybe the files are stored in relationship model but the model has no Nova resource. In this case the link will give 404 error.

The idea of `<a>` tag does not follow the Nova UI, I mean would be better to have an icon SVG like the view one that is used to view a resource detail page.

**Comment by andresilvagomez** [comment]
Captura de Pantalla 2020-12-06 a la(s) 10 54 01 a  m

Maybe a link something like that?

**Comment by ahmedkandel** [comment]
Yes the eye icon is better but still you need to check if the relationship resource exists.

**Comment by andresilvagomez** [comment]
Righ but the icon is to show the document in the navigation according with the other one PR 

In this case I represented it like a relationship with link in the name

**Comment by ahmedkandel** [comment]
I think the eye icon is better for the show resource details while for the open in browser PR use this SVG https://github.com/tailwindlabs/heroicons/blob/master/optimized/outline/external-link.svg

File: resources/js/components/FileCard.vue [hunk 1]
```diff
@@ -1,11 +1,15 @@
 <template>
     <card class="flex item-center overflow-hidden p-4">
-        <div
-            ref="fileIcon"
-            v-tooltip.click="formatedFileMeta"
-            class="flex items-center flex-no-shrink mr-4"
+        <a
```
Comment (ahmedkandel) [inline resources/js/components/FileCard.vue]: Better use `<icon type="download" view-box="0 0 24 24" width="16" height="16" />` to align with Nova UI

File: src/Http/Controllers/FilesController.php [hunk 2]
```diff
@@ -93,6 +93,13 @@ private function files()
                     $data['fileMeta'][$key] = $file[$column] ?? null;
                 }
 
+                if ($this->tool->relationship === 'HasMany' || $this->tool->relationship === 'HasMany') {
+                    $base = config('nova.url') . config('nova.path');
+                    $url = url("$base/resources/{$this->tool->attribute}/{$file['id']}");
```
Comment (ahmedkandel) [inline src/Http/Controllers/FilesController.php:98]: Better to return `$data['modelId']` then compose the url on the front-end

File: src/Http/Controllers/FilesController.php [hunk 3]
```diff
@@ -93,6 +93,13 @@ private function files()
                     $data['fileMeta'][$key] = $file[$column] ?? null;
                 }
 
+                if ($this->tool->relationship === 'HasMany' || $this->tool->relationship === 'HasMany') {
```
Comment (ahmedkandel) [inline src/Http/Controllers/FilesController.php]: Duplicated  `$this->tool->relationship === 'HasMany'` maybe also HasOne can be useful so will be `if ($this->tool->relationship)`

File: src/NovaS3MultipartUpload.php [hunk 4]
```diff
@@ -528,6 +529,17 @@ public function displayPoweredByUppy($boolean = true)
         return $this->withMeta(['proudlyDisplayPoweredByUppy' => $boolean]);
     }
 
+    /**
+     * View resource when user press click at the file icon.
+     *
+     * @param  bool  $boolean
+     * @return $this
+     */
+    public function viewResource($boolean = true)
```
Comment (ahmedkandel) [inline src/NovaS3MultipartUpload.php]: Better to name it `linkToResource` and you can check here if the relationship model has a corresponding Nova resource

**Review by ahmedkandel — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/FilesController.php (hunk 2 above)
Comment (andresilvagomez) [reply src/Http/Controllers/FilesController.php:98]: I'm not sure how do it in the frontend!!!

**Comment by andresilvagomez** [comment]
@ahmedkandel changes are ready

**Comment by andresilvagomez** [comment]
@ahmedkandel

**Comment by ahmedkandel** [comment]
Hey, @andresilvagomez I am thinking about a much wider usage of this feature. Not only a link or a button to model resource But a feature/method to allow the user to create links or buttons e.g. `->linkTo('/nova/videos/{id}')` then the package replace `{id}` with the model id we may also support other routes paramters. Also, think about support for Vue Router in order not to reload pages if the link goes to an internal nova route.

What do you think?

**Comment by andresilvagomez** [comment]
@ahmedkandel I agree with you :) totally, but then could you release a new tag with the last PR, and I will work in this feature today!!

**Comment by ahmedkandel** [comment]
@andresilvagomez no need for new releases just sync your fork with the upstream. Please close this PR and create a new clean one with the updated feature.

**Comment by andresilvagomez** [comment]
> @andresilvagomez no need for new releases just sync your fork with the upstream. Please close this PR and create a new clean one with the updated feature.

@ahmedkandel no is really create a new release tag. to start using new features available in the projects

**Comment by ahmedkandel** [comment]
@andresilvagomez you can start using the new features without waiting for a release by installing the **dev-master** branch as following `composer require "ahmedkandel/nova-s3-multipart-upload:master"`.

We are currently working on other updates and once they are ready will release a new version.

**Comment by ahmedkandel** [comment]
> @ahmedkandel I agree with you :) totally, but then could you release a new tag with the last PR, and I will work in this feature today!!

@andresilvagomez are you working already on this or shall I?

**Pull request closed without merging**